src/ceph-volume/ceph_volume/devices/lvm/listing.py

```diff
@@ -69,51 +68,41 @@ def direct_report():
     bypasses the need to deal with the class interface which is meant for cli
     handling.
     """
+    # TODO: code here looks fishy; we can simplify perhaps.
     _list = List([])
-    # this is crucial: make sure that all paths will reflect current
-    # information. In the case of a system that has migrated, the disks will
-    # have changed paths
-    _list.update()
     return _list.full_report()
 
 
+# TODO: move somewhere in our codebase where it's more accessible/usable?
+def is_ceph_device(lv):
+    try:
+        lv.tags['ceph.osd_id']
+    except (KeyError, AttributeError):
+        logger.warning('device is not part of ceph: %s', lv)
+        return False
+
+    return True
+
+# TODO: move somewhere in our codebase where it's more accessible/usable?
+def get_device_type(lv):
+    for device_type in ['journal', 'block', 'wal', 'db']:
+        if lv.tags.get('ceph.%s_device' % device_type):
+            return device_type
+    else:
+        return 'lv'
```

Comment:
I think this should a a method of `Volume` (see https://github.com/ceph/ceph/blob/d88ff292fe2b092d5588cb82db45434f65224a33/src/ceph-volume/ceph_volume/api/lvm.py#L889)

Reply:
Yes, that would be better, I suppose.